apps/vmq_diversity/src/vmq_diversity_plugin.erl

```diff
@@ -342,10 +384,65 @@ auth_on_subscribe_m5(UserName, SubscriberId, Topics, Props) ->
                                                      {client_id, ClientId},
                                                      {topics, [[unword(T), [QoS, Unmap(SubOpts)]]
                                                                || {T, {QoS, SubOpts}} <- Topics]},
-                                                     {properties, maps:to_list(Props)}]),
+                                                     {properties, conv_args_props(Props)}]),
             conv_res(auth_on_sub, Res)
     end.
 
+on_subscribe_m5(UserName, SubscriberId, Topics, Props) ->
+    {MP, ClientId} = subscriber_id(SubscriberId),
+    Unmap = fun(SubOpts) ->
+                    vmq_diversity_utils:unmap(SubOpts)
+            end,
+    all(auth_on_subscribe_m5, [{username, nilify(UserName)},
```

Comment:
shouldn't this bie `on_subscribe_m5` instead of `auth_on_subscribe_m5`

Reply:
Fixed this and also the test - too much cut and paste :/